bazel/envoy_mobile_repositories.bzl

```diff
@@ -52,10 +52,10 @@ def upstream_envoy_overrides():
     http_archive(
         name = "com_google_absl",
         patches = ["@envoy_mobile//bazel:abseil.patch"],
```

Comment:
Can we remove this patch now? Removes linking Foundation on macOS, but that should be safe? https://github.com/envoyproxy/envoy-mobile/blob/main/bazel/abseil.patch

Reply:
just pushed up a commit to try that out

Reply:
Ah it looks like https://github.com/abseil/abseil-cpp/issues/326 is still open and this could still be a problem.

Reply:
If we can indeed remove the patch, that means we can remove this dep override altogether and implicitly use the one defined in Envoy.

Reply:
Given that the issue is still open, it would seem we should not remove the patch.  But I would assume CI would catch it?  (it's still running).  If we inherit from the Envoy dependencies by default and this is just overrides, then yes, would be great to remove this all together!  I'll wait to see if CI passes.  If it does, I'll remove both the patch and the dependency override.

Reply:
I'm a bit concerned that because we build the aar on a macOS host and run the app verifications on a macOS host too, that it could mask the issue that could show up if we ran the aar on an emulator on a non-Darwin host (i.e. Linux) or on a physical Android device.

Reply:
I think what I'll do is put the patch back in, and add a comment referencing the open GH issue, so people know to keep the patch around until the GH issue is resolved.

Reply:
There's already a link in a comment above, that's how I knew to check. But yes, let's be safe and re-add the patch.

Reply:
Done